cmd/containerd-shim/main_unix.go

```diff
@@ -136,23 +139,37 @@ func serve(server *grpc.Server, path string) error {
 	return nil
 }
 
-func handleSignals(signals chan os.Signal, server *grpc.Server) error {
-	for s := range signals {
-		logrus.WithField("signal", s).Debug("received signal")
-		switch s {
-		case unix.SIGCHLD:
-			if err := reaper.Reap(); err != nil {
-				logrus.WithError(err).Error("reap exit status")
+func handleSignals(signals chan os.Signal, server *grpc.Server, sv *shim.Service) error {
```

Comment:
I think it's mostly to avoid having
```
handleSignals()
return nil
```

The initial code never returned an error.

Happy to change it.